mm2src/mm2_event_stream/src/event.rs

```diff
@@ -0,0 +1,45 @@
+use serde_json::Value as Json;
+
+// Note `Event` shouldn't be `Clone`able, but rather Arc/Rc wrapped and then shared.
+// This is only for testing.
+#[cfg_attr(test, derive(Clone, Debug, PartialEq))]
+#[derive(Default)]
+/// Multi-purpose/generic event type that can easily be used over the event streaming
+pub struct Event {
+    /// The type of the event (balance, network, swap, etc...).
+    event_type: String,
+    /// The message to be sent to the client.
+    message: Json,
+    /// Indicating whether this event is an error event or a normal one.
+    error: bool,
+}
+
+impl Event {
+    /// Creates a new `Event` instance with the specified event type and message.
+    #[inline]
+    pub fn new(streamer_id: String, message: Json) -> Self {
+        Self {
+            event_type: streamer_id,
+            message,
+            error: false,
+        }
+    }
+
+    /// Create a new error `Event` instance with the specified error event type and message.
+    #[inline]
+    pub fn err(streamer_id: String, message: Json) -> Self {
+        Self {
+            event_type: streamer_id,
```

Comment:
I guess we do not use the standard SSE [event type](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event) field (which could be used in addEventListener) and this event_type goes just into the 'data' field.
Did we ever consider use meaningful constants as event types?
I could imaging like: "balance", "trezor-init", "withdraw" etc, so GUI could use them in client scripts.
(LMK if this was discussed already)

Reply:
> I guess we do not use the standard SSE [event type](https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event) field (which could be used in addEventListener) and this event_type goes just into the 'data' field.

I thought i had a todo for this somewhere but i couldn't find it now. Will add it then.
Aha we don't use the event type field, which might be beneficial for the GUI. I think i only wanted to wait and see if they need it one way or another when they start using this.

> I could imaging like: "balance", "trezor-init", "withdraw" etc, so GUI could use them in client scripts.
(LMK if this was discussed already)

tbh, i specifically didn't want to let the GUIs construct the event types themselves, but rather get the types from the response of the enable request instead and remember that this event type is for such and such.
the meaningful constants technique would work but only as long as we really keep them constants. any later changes in these names or how they are constructed has to be done in the GUI as well.

Reply:
Okay just to remember about this. Maybe GUI would like to have it in future